### PR TITLE
Allow to attach files to news

### PR DESCRIPTION
**Pull request by thomasdelhomenie**
Allow to attach files to news

### CODE REVIEW

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      const sizeMB = fileSize / this.BYTES_IN_MB;
+      return sizeMB.toFixed(formattedSizePrecision);
+    },
+    getNewUploadId: function() {
+      const maxUploadId = 100000;
+      return Math.floor(Math.random() * maxUploadId);
+    },
+    queueUpload: function(file) {
+      if(this.value.length >= this.maxFilesCount) {
+        console.log('Max number of attached files reached!');
```
Comment (aelbenney) [inline webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: To be deleted!

Comment (thomasdelhomenie) [reply webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: Done

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+          file.uploadProgress = Math.round(progress.loaded * this.maxProgress / progress.total);
+        }
+      }).then(() => {
+
+        // Check if the file has correctly been uploaded (progress=100) before refreshing the upload list
+        const progressUrl = `${eXo.env.server.context}/upload?action=progress&uploadId=${file.uploadId}`;
+        fetch(progressUrl)
+          .then(response => response.text())
+          .then(responseText => {
+            // TODO fix malformed json from upload service
```
Comment (aelbenney) [inline webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: Is the fix for malformed json done here [https://github.com/exo-addons/news/pull/109/files#diff-4d09dda5c12ce8a749aa598deab9450aR226](url) ?
If yes we can delete this comment.

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+  transition: visibility 0s 3s, opacity 2s 1s linear;
+}
+
+.fade-leave-active {
+  transition: opacity .6s;
+}
+.fade-leave-to {
+  opacity: 0;
+}
+</style>
```
Comment (aelbenney) [inline webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: To be moved to news.less.

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    <exo-document-selector v-if="showDocumentSelector" @close="showDocumentSelector = false" @cancel="showDocumentSelector = false"></exo-document-selector>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  props: {
+    value: {
```
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: `value` is the variable name used by default by vue.js to bind with the v-model. I tried to customize it, as documented in https://fr.vuejs.org/v2/guide/components-custom-events.html#Personnaliser-le-v-model-du-composant but it did not work. I don't know why.

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue (hunk 2 above)
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: No, this line is the trick, not the fix. The fix should be done in the UploadService directly, in gatein.

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue (hunk 3 above)
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/news-activity-composer-app/components/ExoFilesSelector.vue]: Done

**Pull request merged**